cdap-app-fabric/src/main/java/co/cask/cdap/metadata/MetadataServiceModule.java

```diff
@@ -37,5 +51,18 @@ protected void configure() {
     handlerBinder.addBinding().to(LineageHandler.class);
 
     bind(MetadataAdmin.class).to(DefaultMetadataAdmin.class);
+    if (changePublishingRequired) {
```

Comment:
I think this should be restructured a bit. There are two different cases here -
   1. Users do not want to publish metadata (I think you are planning on handling that in a later PR).
   2. No metadata publishing for in-memory and standalone mode.

For point 1, we need to have a config parameter that will be used to control the publishing.

For point 2, we can directly bind `MetadataChangePublisher` to `NoOpMetadataChangePublisher` in `getSingleNodeModules()` and `getInMemoryModules()`, after extending  `MetadataServiceModule` from `RuntimeModule`